tests/testthat/test-misc.R

```diff
@@ -0,0 +1,73 @@
+
+
+
+
+
+
+test_that("rbmi can handle 'grouped' dplyr data", {
+
+    dat <- simulate_data(100) %>%
+        as_tibble() %>%
+        mutate(outcome = if_else(rbinom(n(), 1, 0.2) == 1, NA_real_, outcome))
+
+    dat <- expand_locf(
+        dat,
+        id = levels(dat$id),
+        visit = levels(dat$visit),
+        vars = c("age", "group", "sex"),
+        group = c("id"),
+        order = c("id", "visit")
+    )
+
+    dat_ice <- dat %>%
+        arrange(id, visit) %>%
+        filter(is.na(outcome)) %>%
+        group_by(id) %>%
+        slice(1) %>%
+        ungroup() %>%
+        select(id, visit) %>%
+        mutate(strategy = "JR")
+
+    vars <- set_vars(
+        outcome = "outcome",
+        visit = "visit",
+        subjid = "id",
+        group = "group",
+        covariates = c("sex*age", "visit*group")
+    )
+
+    method <- method_bayes(
+        burn_in = 200,
+        burn_between = 3,
+        n_samples = 250,
+        verbose = FALSE
+    )
+
+    set.seed(987)
+    drawObj <- draws(
+        data = dplyr::group_by(dat, id),
+        data_ice = dplyr::group_by(dat_ice, id),
+        vars = vars,
+        method = method
+    )
+
+    imputeObj <- impute(drawObj, references = c("A" = "A", "B" = "A"))
+
+    vars2 <- vars
+    vars2$covariates <- c("sex*age")
+    anaObj <- analyse(imputeObj, vars = vars2)
+    poolObj <- pool(anaObj)
+
+    devnull <- capture.output({
+        print(drawObj)
+        print(imputeObj)
+        print(anaObj)
+        print(poolObj)
+    })
+    expect_true(length(devnull) >= 1)
```

Comment:
I wasn't really sure what to put in the  expect command, basically I just wanted to run through everything end to end and make sure it doesn't through an error at any point, the point here was just to have something that would be always true unless the prior program had errored.   I wasn't sure if there was a better way of doing it though.

Reply:
I see your point. If I understood it correctly, should the result be the same if I run rbmi on the same data without using "group_by"? Or what exactly happens when running rbmi on grouped data otherwise?

Reply:
It should be the same regardless of wether its grouped or not.